api/v1/swagger/paths/holdings.json

```diff
@@ -0,0 +1,352 @@
+          "type": "string"
+        },
+        {
+          "name": "ccode",
+          "in": "query",
+          "description": "Case insensitive starts-with search for authorized value for the collection code for this holding",
+          "type": "string"
+        },
+        {
+          "name": "callnumber",
+          "in": "query",
+          "description": "Case insensitive starts-with search for call number for this holding",
+          "type": "string"
+        },
+        {
+          "name": "suppress",
+          "in": "query",
+          "description": "Case insensitive starts-with search for boolean indicating whether the record is suppressed in OPAC",
+          "type": "integer"
+        },
+        {
+          "name": "timestamp",
+          "in": "query",
+          "description": "Case insensitive starts-with search for date and time this recird was last altered",
+          "type": "string"
+        },
+        {
+          "name": "datecreated",
+          "in": "query",
+          "description": "Case insensitive starts-with search for date this record was added to Koha",
+          "type": "string"
+        },
+        {
+          "name": "marcflavour",
+          "in": "query",
+          "description": "Case insensitive starts-with search for marcflavour",
+          "type": "string"
+        },
+        {
+          "name": "format",
+          "in": "query",
+          "description": "Case insensitive starts-with search for format",
+          "type": "string"
+        },
+        {
+          "name": "deleted_on",
+          "in": "query",
+          "description": "Case insensitive starts-with search for date this record was deleted from Koha",
+          "type": "string"
+        }
+      ],
+      "produces": ["application/json"],
+      "responses": {
+        "200": {
+          "description": "A list of holdings",
+          "schema": {
+            "$ref": "../definitions.json#/holdings"
+          }
+        },
+        "404": {
+          "description": "No holdings found",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "500": {
+          "description": "Internal server error",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "503": {
+          "description": "Under maintenance",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        }
+      }
+    },
+    "post": {
+      "x-mojo-to": "Holding#add",
+      "operationId": "addHolding",
+      "tags": ["patrons", "holdings"],
+      "parameters": [{
+        "name": "body",
+        "in": "body",
+        "description": "A JSON object containing informations about the new holding",
+        "required": true,
+        "schema": {
+          "$ref": "../definitions.json#/holding"
+        }
+      }
+      ],
+      "consumes": ["application/json"],
```

Comment:
This doesn't appear to be right since the code processed XML

Reply:
Good that you point this out. I suggest we follow Koha [Biblios endpoint RFC](https://wiki.koha-community.org/wiki/Biblios_endpoint_RFC). It states

> Several serialization formats and schemas could be supported

https://wiki.koha-community.org/wiki/Biblios_endpoint_RFC#Metadata_format_handling

In order to support this, we need to backport Bug 21334
https://bugs.koha-community.org/bugzilla3/show_bug.cgi?id=21334

Reply:
Done. Force pushed plenty of new code.

POST & PUT now consumes
        "application/marcxml+xml",
        "application/marc-in-json",
        "application/marc"